### PR TITLE
Add devfile 2.0 devfile for che-theia

### PR DESCRIPTION
**Pull request by ericwill**
Signed-off-by: Eric Williams <ericwill@redhat.com>

### What does this PR do?
Add the devfile 2.0 definition of the che-theia devfile

### What issues does this PR fix or reference?

CRW-975

### CODE REVIEW

File: dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/7.28.3/devfile.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  - id: init-container-command
+    apply:
+      component: remote-runtime-injector
+events:
+  preStart:
+    - init-container-command
+components:
+  - name: theia-ide
+    container:
+      image: 'quay.io/crw/theia-rhel8:2.8-6": "de36fdf140ba6367e6edf577d6dbaffa270e5e5ecf0890e498f5907f8287858f'
```
Comment (benoitf) [inline dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/7.28.3/devfile.yaml:14]: format looks invalid there no?

Comment (benoitf) [reply dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/7.28.3/devfile.yaml:14]: also @nickboldt can we use "registry.redhat.io/codeready-workspaces/theia-rhel8:2.8" and it'll be replaced automatically by a release script ? or we need to use full digest

Comment (nickboldt) [reply dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/7.28.3/devfile.yaml:14]: yeah, that looks wrong... should it be `quay.io/crw/theia-rhel8@sha256:de36fdf140ba6367e6edf577d6dbaffa270e5e5ecf0890e498f5907f8287858f` ?

Comment (benoitf) [reply dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/7.28.3/devfile.yaml:14]: https://issues.redhat.com/secure/attachment/12519959/12519959_crw-theia-devfile.yaml

Comment (nickboldt) [reply dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/7.28.3/devfile.yaml:14]: > can we use "registry.redhat.io/codeready-workspaces/theia-rhel8:2.8"

I'll need to verify how this works in the dockerfile ... not sure offhand which we put in there.

File: dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/next/devfile.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  - id: init-container-command
+    apply:
+      component: remote-runtime-injector
+events:
+  preStart:
+    - init-container-command
+components:
+  - name: theia-ide
+    container:
+      image: 'quay.io/crw/theia-rhel8@sha256:de36fdf140ba6367e6edf577d6dbaffa270e5e5ecf0890e498f5907f8287858f'
```
Comment (benoitf) [inline dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/next/devfile.yaml:14]: in the CRW issue, registry.redhat.io was used

Comment (ericwill) [reply dependencies/che-plugin-registry/v3/plugins/eclipse/che-theia/next/devfile.yaml]: Fixed it

**Review by benoitf — APPROVED** [review]
@sleshchenko you confirm we only need devfile.yaml and not entry in index.json like https://che-plugin-registry-main.surge.sh/v3/plugins/index.json there ?

**Comment by sleshchenko** [comment]
> @sleshchenko you confirm we only need devfile.yaml and not entry in index.json like che-plugin-registry-main.surge.sh/v3/plugins/index.json there ?

@benoitf yeah. Dashboard does not use devfile.yaml yet https://github.com/eclipse-che/che-dashboard/blob/264ff04c817ecb5fdf642cdb5b8b5f3b8c35e2e5/src/services/bootstrap/PreloadData.ts#L130
And we tested it worked without changing in index.json just fine.

**Comment by benoitf** [comment]
@nickboldt so whenever you want to do a respin, please merge first :-)

**Pull request merged**